src/sdk.js

```diff
@@ -1,57 +1,156 @@
 window.Smartcar = (function(window) {
   'use strict';
+  // accesss denied error class
+  class AccessDenied extends Error { // eslint-disable-line no-restricted-syntax
+    constructor(message) {
+      super(message);
+      this.name = 'AccessDenied';
+    }
+  }
+
+  /**
+   * @typedef Options
+   *
+   * @property {String} clientId - the application's client id
+   * @property {String} redirectUri - the registered redirect uri of the
+   * application
+   * @property {String[]} scope - requested permission scopes
+   * @property {Function} onComplete - called on completion of auth flow
```

Comment:
nit: this prop is optional

also this might be overkill so feel free to ignore, but thoughts on typedef'ing the `oncomplete` using `@callback`?